packages/invoices/src/OutboundShipment/DetailView/tabs/GeneralTab.tsx

```diff
@@ -117,13 +112,23 @@ export const GeneralTabComponent: FC<
             canExpand: false,
           });
         });
-      }
-    });
-    setGrouped(newGrouped);
-  }, [isOn, data]);
+        setGrouped(newGrouped);
+      });
+    }
+  }, [isGroupedByItem, data]);
 
   const t = useTranslation('distribution');
-  const activeRows = grouped.filter(({ isDeleted }) => !isDeleted);
+  const activeRows = useMemo(
+    () => grouped.filter(({ isDeleted }) => !isDeleted),
+    [grouped]
+  );
+  const { setIsGrouped } = useTableStore();
```

Comment:
The column header isn't showing when navigating with the persisted `isGrouped` as true

A little effect should handle it

```
useEffect(() => {
    setIsGrouped(storedIsGrouped);
  }, [storedIsGrouped, setIsGrouped]);
```

Could then also remove this from the toggle fn also, I think